### PR TITLE
Shanoir issue#647 Download management

### PR DESCRIPTION
**Pull request by jcomedouteau**
- Better rights management on examination, study, datasets APIs
- Remove double download button
- Set mime type
- default download to nii

### CODE REVIEW

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java [hunk 1]
```diff
@@ -161,30 +161,31 @@
...
 	@ApiResponses(value = {
-			@ApiResponse(code = 204, message = "examination updated"),
+	        @ApiResponse(code = 200, message = "protocol file", response = Resource.class),
 			@ApiResponse(code = 401, message = "unauthorized", response = Void.class),
 			@ApiResponse(code = 403, message = "forbidden", response = Void.class),
 			@ApiResponse(code = 422, message = "bad parameters", response = ErrorModel.class),
 			@ApiResponse(code = 500, message = "unexpected error", response = ErrorModel.class) })
 	@GetMapping(value = "protocol-file-download/{studyId}/{fileName:.+}/")
-	@PreAuthorize("hasRole('ADMIN')")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @studySecurityService.hasRightOnStudy(#studyId, 'CAN_SEE_ALL'))")
```
Comment (michaelkain) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java:186]: Hi Jean-Come, could you please change this to can_download? every user has can_see_all from default, when member of the study. From my point of view the idea is, that with can_see users can look on everything on the web, but can not download/"take away" contents. I hope this sounds logic. With kind regards, Michael

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java [hunk 2]
```diff
@@ -209,6 +210,7 @@ void downloadProtocolFile(
         @ApiResponse(code = 404, message = "no dataset found"),
         @ApiResponse(code = 500, message = "unexpected error", response = ErrorModel.class) })
     @GetMapping(value = "/exportBIDS/studyId/{studyId}")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @studySecurityService.hasRightOnStudy(#studyId, 'CAN_SEE_ALL'))")
```
Comment (michaelkain) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java:213]: same for downloading BIDS tree for me

**Review by michaelkain — APPROVED** [review]
very good! please see just my two minor comments

**Comment by jcomedouteau** [comment]
Well seen ! It's done ;)

**Pull request merged**